File: problem/mult/6/main.lua

-- main.lua: main file for "mult" solutions
--
-- Copyright (C) 2020-2021  Rudy Matela
-- All rights reserved
--
--
-- This is appended to the submitted Lua program and tests the add function.
--
-- The submitted file processes the standard input and this processes the
-- "in.txt" file.

-- standard input is already processed
-- as the submitted programmed is run when imported
io.input("in.txt")
x, y, z = io.read("*n", "*n", "*n")
while x and y and z do
  print(mult(x,y,z))
  x, y, z = io.read("*n", "*n", "*n")
end

File: problem/mult/6/main.cc
/*
 * main.c: main file for "mult" solutions
 *
 * Copyright (C) 2020-2021  Rudy Matela
 * All rights reserved.
 *
 *
 * This file is part of the CSCX Online Judge.
 *
 * This is to be linked to the submitted file.
 * It processes values from standard input then from the "in.txt" file.
 */
#include <stdio.h>
int mult(int, int, int);
int main()
{
	int i, j, k;
	FILE *f;
	/* process standard input */
	while (scanf(" %d %d %d", &i, &j, &k)==3)
		printf("%d\n", mult(i,j,k));
	/* process additional inputs from in.txt */
	f = fopen("in.txt", "r");
	if (!f) {
		fprintf(stderr, "could not open in.txt");
		return 1;
	}
	while (fscanf(f," %d %d %d", &i, &j, &k)==3)
		printf("%d\n", mult(i,j,k));
	fclose(f);
	return 0;
}
